src/main/scala/firrtl/passes/InferBinaryPoints.scala

```diff
@@ -0,0 +1,103 @@
+// See LICENSE for license details.
+
+package firrtl.passes
+
+// Datastructures
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.immutable.ListMap
+
+import firrtl._
+import firrtl.ir._
+import firrtl.Utils._
+import firrtl.Mappers._
+import firrtl.Implicits.{constraint2width, width2constraint}
+
+class InferBinaryPoints extends Pass {
+  private val constraintSolver = new ConstraintSolver()
+
+  private def addTypeConstraints(t1: Type, t2: Type): Unit = (t1,t2) match {
```

Comment:
I see warnings in compile on the complete-ness of the match at line 18. Perhaps add a catch-all?